src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -257,6 +266,29 @@ private async Task ConnectAsync(CancellationToken token = default(CancellationTo
         }
     }
 
+    private void PrepareConnection()
+    {
+        ValidateServerFeatures();
+        TryRegisterProfiler();
+    }
+
+    private void ValidateServerFeatures()
+    {
+        if (_connection is null)
```

Comment:
Is `_connection` ever really null here? Maybe just assert if not.

Reply:
Made by analogy with existing [TryRegisterProfiler()](https://github.com/dotnet/aspnetcore/blob/a364bcbd1c360762d5e8bda8fe04fa004973489a/src/Caching/StackExchangeRedis/src/RedisCache.cs#L262).
Imho, it's better to make less noise in likely scenarios.

Reply:
I guess this doesn't make things any worse. I prefer assertions over if checks where possible because it makes future code reviews easier. It makes it much more obvious what the invariants actually are.

Reply:
Clear.
Should I change the check to assertion in [TryRegisterProfiler()](https://github.com/dotnet/aspnetcore/blob/a364bcbd1c360762d5e8bda8fe04fa004973489a/src/Caching/StackExchangeRedis/src/RedisCache.cs#L262) method for consistency?

Reply:
If you don't mind. This was just a nit on my part. Thanks!